cpp/ql/src/Security/CWE/CWE-190/TaintedAllocationSize.ql

```diff
@@ -13,21 +13,33 @@
 
 import cpp
 import semmle.code.cpp.security.TaintTracking
+import TaintedWithPath
 
-predicate taintedAllocSize(Expr e, Expr source, string taintCause) {
+predicate taintedChild(Expr e, Expr tainted) {
   (
-    isAllocationExpr(e) or
+    isAllocationExpr(e)
+    or
     any(MulExpr me | me.getAChild() instanceof SizeofOperator) = e
   ) and
+  tainted = e.getAChild() and
```

Comment:
Do you mean for a `NewExpr`? I don't think that will happen because the constructor call [is a child of the `NewExpr`](https://github.com/Semmle/ql/blob/316d932829fbea6c55fbe390e3b8c5951ec2321b/cpp/ql/src/semmle/code/cpp/exprs/Expr.qll#L841), and a constructor _argument_ is therefore a grandchild.